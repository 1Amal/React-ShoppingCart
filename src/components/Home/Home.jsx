import { Link } from "react-router-dom";

import HomeCSS from "./Home.module.css";
import Navigation from "../Navigation/Navigation";
function Home() {
  return (
    <div className={HomeCSS.container}>
      <Navigation />
      <h1>Welcome to Mock Shopping cart developed using React</h1>
      <div className={HomeCSS.container}>
        <h2>About</h2>
        <ul className={HomeCSS.list}>
          <li>
            <a href="https://github.com/1Amal/React-ShoppingCart">
              Source Code
            </a>
          </li>
          <li>
            <a href="https://amalk-reactshoppingcart.netlify.app/">Live Demo</a>
          </li>
        </ul>
      </div>
      <div className={HomeCSS.container}>
        <h2 className={HomeCSS.headingsH1}>App Architecture</h2>
        <h3>Component Structure</h3>
        <ul className={HomeCSS.list}>
          <li className={HomeCSS.listItem}>Main =&gt; App =&gt; Home =&gt; </li>
          <li className={HomeCSS.listItem}>Shop =&gt; </li>
          <li className={HomeCSS.listItem}>Shopping Cart</li>
          <li className={HomeCSS.listItem}>Product</li>
        </ul>
        <div className={HomeCSS.container}>
          <h2 className={HomeCSS.headingsH2}>App Specification</h2>
          <ul className={HomeCSS.list}>
            <li className={HomeCSS.listItem}>Consistent Navigation Bar</li>
            <li>Fetch shop items from FakeStore API </li>
            <li>Clear out any missing in props validation errors</li>
            <li>Use React Testing Library to test the App</li>
            <li>Style the app using CSS Modules and Styled Components</li>
          </ul>
        </div>
      </div>
    </div>
  );
}

export default Home;
