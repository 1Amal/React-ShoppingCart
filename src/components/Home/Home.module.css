.container {
  display: flex;
  flex-direction: column;
  padding: 0rem 0rem 5rem 0rem;
  margin: 0rem 0rem 1rem 0rem;
}

.headingsH1 {
  padding-bottom: 2rem;
}

.list {
  background-color: rgb(255, 255, 255);
  /* list-style-type: circle; */
  list-style: none;
  padding: 0rem;
}

.listItem {
  padding: 0rem;
}
