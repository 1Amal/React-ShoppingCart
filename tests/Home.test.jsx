import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Home from "../src/components/Home/Home";

describe("Home Component", () => {
  it("matches the snapshot", () => {
    const { container } = render(
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
});