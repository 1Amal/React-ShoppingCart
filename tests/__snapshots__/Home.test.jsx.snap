// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home Component > matches the snapshot 1`] = `
<div>
  <div
    class="_container_345077"
  >
    <div
      class="_navigationBar_70c253"
    >
      <ul>
        <li>
          <a
            href="/home"
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/shop/cart"
          >
            Shop
          </a>
        </li>
        <li>
          <a
            href="/shop/cart"
          >
            Cart 
            <span>
              <strong>
                (
                0
                )
              </strong>
            </span>
          </a>
        </li>
      </ul>
    </div>
    <h1>
      Welcome to Mock Shopping cart developed using React
    </h1>
    <div
      class="_container_345077"
    >
      <h2>
        About
      </h2>
      <ul
        class="_list_345077"
      >
        <li>
          <a
            href="https://github.com/1Amal/React-ShoppingCart"
          >
            Source Code
          </a>
        </li>
        <li>
          <a
            href="https://amalk-reactshoppingcart.netlify.app/"
          >
            Live Demo
          </a>
        </li>
      </ul>
    </div>
    <div
      class="_container_345077"
    >
      <h2
        class="_headingsH1_345077"
      >
        App Architecture
      </h2>
      <h3>
        Component Structure
      </h3>
      <ul
        class="_list_345077"
      >
        <li
          class="_listItem_345077"
        >
          Main =&gt; App =&gt; Home =&gt; 
        </li>
        <li
          class="_listItem_345077"
        >
          Shop =&gt; 
        </li>
        <li
          class="_listItem_345077"
        >
          Shopping Cart
        </li>
        <li
          class="_listItem_345077"
        >
          Product
        </li>
      </ul>
      <div
        class="_container_345077"
      >
        <h1
          class="_headingsH1_345077"
        >
          App Specification
        </h1>
        <ul
          class="_list_345077"
        >
          <li
            class="_listItem_345077"
          >
            Consistent Navigation Bar
          </li>
          <li>
            Fetch shop items from FakeStore API 
          </li>
          <li>
            Clear out any missing in props validation errors
          </li>
          <li>
            Use React Testing Library to test the App
          </li>
          <li>
            Style the app using CSS Modules and Styled Components
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
